src/main/java/randoop/resource/generator/MethodListGen.java

```diff
@@ -0,0 +1,173 @@
+package randoop.resource.generator;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import annotator.specification.IndexFileSpecification;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.plumelib.reflection.Signatures;
+import scenelib.annotations.Annotation;
+import scenelib.annotations.el.AClass;
+import scenelib.annotations.el.AMethod;
+import scenelib.annotations.el.AScene;
+import scenelib.annotations.io.classfile.ClassFileReader;
+
+public class MethodListGen {
+  private static final String CLASS_EXT = ".class";
+  private static final String JAIF_EXT = ".jaif";
+
+  private static final String NONDET_ANNOTATION =
+      "org.checkerframework.checker.determinism.qual.NonDet";
+  private static final String SEF_ANNOTATION = "org.checkerframework.dataflow.qual.SideEffectFree";
+  private static final String PURE_ANNOTATION = "org.checkerframework.dataflow.qual.Pure";
+
+  private static final List<String> NONDET_QUALIFYING_ANNOTATIONS = new ArrayList<String>();
+  private static final List<String> SEF_QUALIFYING_ANNOTATIONS = new ArrayList<String>();
+
+  static {
+    NONDET_QUALIFYING_ANNOTATIONS.add(NONDET_ANNOTATION);
+    SEF_QUALIFYING_ANNOTATIONS.add(SEF_ANNOTATION);
+    SEF_QUALIFYING_ANNOTATIONS.add(PURE_ANNOTATION); // Pure = SideEffectFree + Deterministic
+  }
+
+  /**
+   * Main entry point to generate Nondeterministic and Side Effect Free Method lists.
+   *
+   * @param args command line arguments as follows: args[0] - root of the input .class directory
+   *     args[1] - root of the input .jaif directory args[2] - output directory for sefMethods.txt,
+   *     and nonDetMethodsRegex.txt.
+   */
+  public static void main(String[] args) {
+    Path classWorkingDirectory = Paths.get(args[0]);
+    Path jaifWorkingDirectory = Paths.get(args[1]);
+
+    Path nonDetFile = Paths.get(args[2] + "/omitmethods-defaults-part.txt");
+    Path sideEffectFreeFile = Paths.get(args[2] + "/JDK-sef-methods.txt");
+
+    try {
+      List<String> nonDetMethods =
+          walkFilesForAnnotation(classWorkingDirectory, CLASS_EXT, NONDET_QUALIFYING_ANNOTATIONS);
+      List<String> sideEffectFreeMethods =
+          walkFilesForAnnotation(jaifWorkingDirectory, JAIF_EXT, SEF_QUALIFYING_ANNOTATIONS);
+
+      // Sort in alphabetical order
+      Collections.sort(nonDetMethods);
+      Collections.sort(sideEffectFreeMethods);
+
+      // Write out the captured methods.
+      System.out.println("Nondeterministic methods count: " + nonDetMethods.size());
+      if (nonDetMethods.size() > 0) {
+        try (BufferedWriter nonDetMethodWriter =
+            Files.newBufferedWriter(nonDetFile.toFile().toPath(), UTF_8)) {
```

Comment:
nonDetFile is already a Path.  Why convert it to a file and then back to a Path?